idseq_dag/util/thread_with_result.py

```diff
@@ -0,0 +1,50 @@
+#!/bin/env python3
+import threading
```

Comment:
Should you be using multiprocessing instead because of PIL?

Reply:
I use multiprocessing where CPU is the bottleneck, and nowhere else :)   Take a look at the `@run_in_subprocess` decorator.   The pattern is:  1) use threads for everything  2) If the CPU is oversubscribed in some section that's critical for performance, run that section in a synchronous subprocess, as though it is a shell command.   I know how to use multiprocessing in its full generality, but prefer to use it in this very restricted manner.

Reply:
Makes, I need to understand the flow better.